src/platform/workflow/core/services/workflowService.ts

```diff
@@ -310,31 +310,37 @@ export const useWorkflowService = () => {
     value: string | ComfyWorkflow | null,
     workflowData: ComfyWorkflowJSON
   ) => {
-    // Use workspaceStore here as it is patched in unit tests.
     const workflowStore = useWorkspaceStore().workflow
-    if (typeof value === 'string') {
-      const workflow = workflowStore.getWorkflowByPath(
-        ComfyWorkflow.basePath + appendJsonExt(value)
-      )
-      if (workflow?.isPersisted) {
-        const loadedWorkflow = await workflowStore.openWorkflow(workflow)
-        loadedWorkflow.changeTracker.restore()
-        loadedWorkflow.changeTracker.reset(workflowData)
-        return
-      }
-    }
 
     if (value === null || typeof value === 'string') {
       const path = value as string | null
+
+      // Check if a persisted workflow with this path exists
+      if (path) {
+        const fullPath = ComfyWorkflow.basePath + appendJsonExt(path)
+        const existingWorkflow = workflowStore.getWorkflowByPath(fullPath)
+
+        // If the workflow exists and is NOT loaded yet (restoration case),
+        // use the existing workflow instead of creating a new one.
+        // If it IS loaded, this is a re-import case - create new with suffix.
+        if (existingWorkflow?.isPersisted && !existingWorkflow.isLoaded) {
+          const loadedWorkflow =
+            await workflowStore.openWorkflow(existingWorkflow)
+          loadedWorkflow.changeTracker.reset(workflowData)
+          loadedWorkflow.changeTracker.restore()
+          return
+        }
+      }
+
       const tempWorkflow = workflowStore.createTemporary(
         path ? appendJsonExt(path) : undefined,
-        workflowData
+        workflowData,
+        true
```

Comment:
Nit: Bare boolean arguments are a code smell, could you convert this to have named parameters / options arg?